server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java

```diff
@@ -214,9 +214,13 @@ public MailQueueItem deQueue() throws MailQueueException {
                     mailQueueSize.decrement();
                     return createMailQueueItem(session, consumer, message);
                 } else {
-                    session.commit();
-                    closeConsumer(consumer);
-                    closeSession(session);
+                    MessageConsumer consumerToClose = consumer;
+                    Session sessionToClose = session;
+                    consumer = null;
```

Comment:
this code doesn't make sense to me, could you explain or write it in a way that I understand it ?

Reply:
Well...

Assumption 1: don't drop the log in the rollback. It might contain useful information. But we want to avoid benign logs, let say during tests...

I want to reduce the time window where 2 close can be triggered...

One is done when naturally handing the mail, one by the interrupt error handling. They happen concurrently.

One good idea is to replace the stuff to close by null (fast operation) so that the other part doesn't attempt to roll it back...

That's the most straightforward way I found to achieve this here...

Reply:
I probably lack some information on the problem you try to solve because I still can understand what you are doing